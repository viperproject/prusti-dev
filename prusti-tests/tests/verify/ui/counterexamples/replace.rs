// compile-flags: -Pcounterexample=true

use prusti_contracts::*;

fn replace(x: &mut char, acc: bool) {
    match x {
        '$' => {
            if acc {
                *x = ' ';
            } else {
               panic!("no access"); 
            }
        }
        _ => {}
    }
}

fn main() {}
