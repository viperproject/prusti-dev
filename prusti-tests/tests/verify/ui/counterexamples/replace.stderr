error: [Prusti: verification error] panic!(..) statement might be reachable
  --> $DIR/replace.rs:11:16
   |
11 |                panic!("no access"); 
   |                ^^^^^^^^^^^^^^^^^^^
   |
note: counterexample for "x"
  initial value: ref('$' (0x24))
  final value:   ref('$' (0x24))
  --> $DIR/replace.rs:5:12
   |
5  | fn replace(x: &mut char, acc: bool) {
   |            ^
note: counterexample for "acc"
  initial value: false
  final value:   false
  --> $DIR/replace.rs:5:26
   |
5  | fn replace(x: &mut char, acc: bool) {
   |                          ^^^
   = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

